### ISSUE TITLE
Check all of the new revision of the CNA Rules once they're official

### ISSUE THREAD

**Issue by andrewpollock** [mprpic/cvelint#13]
I note that Section 5.1 of the updated _CVE Numbering Authority Operational Rules version 4.0_ has a number of MUST statements (I'm not currently sure how much they differ from the current version).

Once this is authoritative, cvelint should (no pun intended) check for all of the MUSTs and perhaps warn on the SHOULDs

**Comment by mprpic** [comment]
Do you know when 4.0 will be available at https://www.cve.org/ResourcesSupport/AllResources/CNARules ? That still points to 3.0 as the most current version.

**Comment by andrewpollock** [comment]
I believe @zmanion is looking to get it ratified in the not too distant future

**Comment by zmanion** [comment]
[This is the final draft](https://docs.google.com/document/d/1YtIjt5EulYuVUBnh2uMgtSus3yDi3iXkg9zJ94TZG9c/), not yet approved by CVE Board.

Here's [the schedule](https://docs.google.com/document/d/1DOb5gLg0hvRMG3sAexL-ycUyw72BHWzM9wc6fqr_EjI/edit#heading=h.2g2frm6ya3br), not likey to change further.

**Comment by mprpic** [comment]
The MUST rules around products:

```
5.1.3 MUST identify at least one affected Product using information such as Supplier and Product names, versions, and dates.
5.1.4 MUST identify at least one Product as “affected” or “unknown” (with the possibility of being affected).
5.1.8 MUST NOT only contain Products with “unaffected” status. This would not be a Vulnerability.
```

are all covered by the E006 check, which detects exactly that at least one products marked as affected or unknown is present.

I'm not exactly sure how this can be checked:

```
5.1.7 MUST identify the type of Vulnerability. The CVE record SHOULD use the Common Weakness Enumeration ([CWE](https://cwe.mitre.org/)) to classify the type or cause of the Vulnerability. A CVE Record MAY contain multiple types or causes of the Vulnerability.
```

since the text doesn't go into detail how exactly the type has to be identified besides a CWE.

```
5.1.9 MUST contain a prose description (see 5.2).
5.1.10 MUST contain at least one public reference (see 5.3).
```
I believe these two are checked at the schema level. There is also a check for an insufficient description (less than 10 characters): E003.

```
5.1.11 MUST include the appropriate and approved CNA tag(s), if applicable. CVE Program members SHOULD present the information provided by tags, including relevant explanations, when rendering and displaying CVE Record data.
5.1.11.1 MUST use the "exclusively-hosted-service" tag when all known Products listed in the CVE Record exist only as fully hosted services. If the Vulnerability affects both hosted services and on-premises Products, then this tag MUST NOT be used.
```
I don't see how we could check either of these programmatically.

```
5.1.12 MUST meet any requirements set by any approved formats or submission mechanisms, for example, the CVE Record Format and the Record Submission and Upload Service (RSUS). Such requirements will be integrated into this document on a periodic basis.
```
This is handled by schema validation itself.

**Comment by mprpic** [comment]
The SHOULDs:

```
5.1.1 SHOULD contain sufficient information to uniquely identify the Vulnerability and distinguish it from similar Vulnerabilities.
5.1.2 SHOULD identify the vulnerable Products, versions, dates, components, sub-components, features, modules, APIs, functions, function calls, commands, utilities, or programs, to the degree necessary to uniquely identify the Vulnerability.
```

This is hard to quantify programmatically since it's difficult to have a good indicator of what makes a vulnerability unique. Is it just the name and the version of the component that it affects along with the CVE ID?

```
5.1.5 SHOULD identify Fixed versions of Products.
```

This is sort of checked by E007 since it'll complain if a fixed version uses "n/a".

**Comment by andrewpollock** [comment]
Yeah I've heard an interesting (to me) interpretation that the "identify" box is checked when that information is in the `description` field 😢 

Having just had my first play with Vulnogram in anger, it enforces a much higher standard than the schema does today, also...

5.1.7 could be checked by examining the [`problemTypes`](https://cveproject.github.io/cve-schema/schema/docs/#oneOf_i0_containers_cna_problemTypes) field, couldn't it? I note that the schema seems to be already requiring *something* be supplied, and so nonsense like [CVE-2024-0042](https://cveawg.mitre.org/api/cve/CVE-2024-0042) prevails...

That said, [CVE-2024-32674](https://cveawg.mitre.org/api/cve/CVE-2024-32674) seems to be an example of doing something more useful.
